prometheus_juju_exporter/config.py

```diff
@@ -0,0 +1,58 @@
+"""Configuration loader."""
+import confuse
+
+from prometheus_juju_exporter.logging import get_logger
+
+config = None
+
+
+class Config:
+    """Configuration for PrometheusJujuExporter."""
+
+    def __init__(self, args=None):
+        """Initialize the config class."""
+        self.logger = get_logger()
+        global config
+
+        if config:
+            self.config = config
+        else:
+            self.config = confuse.Configuration("PrometheusJujuExporter", __name__)
+            config = self.config
+
+        if args:
+            self.config.set_args(args, dots=True)
+
+        self.validate_config_options()
+
+    def validate_config_options(self):
+        expected_config = {
+            "exporter": ["port", "collect_interval"],
+            "juju": [
+                "controller_endpoint",
+                "controller_cacert",
+                "username",
+                "password",
+            ],
+            "customer": ["name", "cloud_name"],
+        }
+
+        try:
+            for section in expected_config.keys():
+                for entry in expected_config[section]:
+                    if section == "exporter" and entry == "port":
+                        self.config[section][entry].as_choice(range(0, 65536))
+                    elif section == "exporter" and entry == "collect_interval":
+                        self.config["exporter"]["collect_interval"].get(int)
+                    else:
+                        self.config[section][entry].get()
+        except Exception as e:
```

Comment:
IMO, we should avoid using `Exception`. Better will be to use `KeyError, confuse.ConfigErrror`